velox/exec/TopN.cpp

```diff
@@ -36,14 +36,30 @@ TopN::TopN(
           topNNode->sortingOrders(),
           data_.get()),
       topRows_(comparator_),
-      decodedVectors_(outputType_->children().size()) {}
+      decodedVectors_(outputType_->children().size()) {
+  sortingKeyColumns_.reserve(topNNode->sortingKeys().size());
+  const auto columnCount{outputType_->children().size()};
+  std::vector<bool> isSortingKey(columnCount);
+  for (const auto& key : topNNode->sortingKeys()) {
+    sortingKeyColumns_.emplace_back(exprToChannel(key.get(), outputType_));
+    isSortingKey[sortingKeyColumns_.back()] = true;
+  }
+  nonKeyColumns_.reserve(columnCount - sortingKeyColumns_.size());
+  for (column_index_t i = 0; i < columnCount; ++i) {
```

Comment:
Skip this loop if number of sorting keys == number of columns.

Reply:
Done.